src/bosh_azure_cpi/lib/cloud/azure/disk/config_disk_manager.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require 'etc'
+
+module Bosh::AzureCloud
+  class ConfigDiskManager
+    include Helpers
+
+    BOSH_CONFIG_DISK_LABEL = 'azure_cfg_dsk'
+    CONFIG_DISK_CONTAINER  = 'azure-config-disks'
+    CONFIG_DISK_MOUNT_POINT = 'azure_config_disk_mount'
+    CONFIG_DISK_FILE_PATH_PREFIX = 'azure_config_disk_image'
+    CONFIGS_RELATIVE_PATH = 'configs'
+    CONFIG_DISK_SIZE = 24 # in MB, azure have a requirement for the min size.
+    MEGA_SIZE = 1024 * 1024
+    def initialize(blob_manager, disk_manager2, storage_account_manager)
+      @blob_manager = blob_manager
+      @disk_manager2 = disk_manager2
+      @storage_account_manager = storage_account_manager
+      @logger = Bosh::Clouds::Config.logger
+    end
+
+    # Returns the config disk uri.
+    def prepare_config_disk(resource_group_name, vm_name, location, metadata_obj, user_data_obj)
+      disk = create_config_disk(resource_group_name, vm_name, location, metadata_obj, user_data_obj)
+      disk
+    end
+
+    private
+
+    def create_config_disk(resource_group_name, vm_name, location, meta_data_obj, user_data_obj)
+      mounted_dir = nil
+      config_disk_file_path = nil
+      config_disk_vhd_file_path = nil
+      disk_name = nil
+      page_blob_created = false
+      command_runner = CommandRunner.new
+      umounted = false
+      begin
+        config_disk_file = Tempfile.new(CONFIG_DISK_FILE_PATH_PREFIX)
+        config_disk_file.close
+        config_disk_file_path = config_disk_file.path
+
+        mk_image_file_cmd = "dd if=/dev/zero of=#{config_disk_file_path} bs=#{MEGA_SIZE} count=#{CONFIG_DISK_SIZE}"
+        command_runner.run_command(mk_image_file_cmd)
+
+        mkfs_cmd = "mkfs.ext4 -F #{config_disk_file_path} -L #{BOSH_CONFIG_DISK_LABEL}"
+        command_runner.run_command(mkfs_cmd)
+
+        mounted_dir = Dir.mktmpdir(CONFIG_DISK_MOUNT_POINT)
+
+        mount_cmd = "sudo mount -o loop #{config_disk_file_path} #{mounted_dir}"
+        command_runner.run_command(mount_cmd)
+
+        user = ENV['USER']
+        chmod_cmd = "sudo chown #{user} #{mounted_dir}"
+        command_runner.run_command(chmod_cmd)
+
+        mkdir_cmd = "mkdir -p #{mounted_dir}/#{CONFIGS_RELATIVE_PATH}"
+        command_runner.run_command(mkdir_cmd)
+
+        meta_data_str = JSON.dump(meta_data_obj)
+        File.open("#{mounted_dir}/#{CONFIGS_RELATIVE_PATH}/MetaData", 'w') do |file|
+          file.write(meta_data_str)
+        end
+
+        user_data_str = JSON.dump(user_data_obj)
+        File.open("#{mounted_dir}/#{CONFIGS_RELATIVE_PATH}/UserData", 'w') do |file|
+          file.write(user_data_str)
+        end
+
+        unmount_cmd = "sudo umount #{mounted_dir}"
+        command_runner.run_command(unmount_cmd)
+        umounted = true
+        # convert the plain file to vhd
+        # TODO check whether qemu-utils is installed or not. if not, install it.
```

Comment:
weird to install it here... can we include the util in stemcell?

Reply:
yes, doing it in this branch https://github.com/andyliuliming/bosh-linux-stemcell-builder/commits/config_disk , hope it can be accepted by bosh team.
if not, then we are going to think about to install it here.